matrix/buffer.py

```diff
@@ -347,6 +355,7 @@ def __init__(self, name, server_name, user):
 
         self.name = ""
         self.users = {}  # type: Dict[str, WeechatUser]
+        self.duplicate_nicks = {} # type: Dict[str, List[WeechatUser]]
```

Comment:
You can use a defaultdict here, that way we can always use duplicate_nicks[nick].append().